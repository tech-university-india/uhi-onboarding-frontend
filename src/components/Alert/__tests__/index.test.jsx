import React from 'react'
import Alert from '..'
import { fireEvent, render } from '@testing-library/react'
describe('Tests for Alert Dialog', () => {
  it('should render the component', () => {
    const wrapper = render(<Alert />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render the component with props', () => {
    const screen = render(<Alert input="test" />)
    expect(screen.getByText('test')).toBeTruthy()
  })
  it('should call the close alert callback when the close button is clicked', () => {
    const setAlert = jest.fn()
    const screen = render(<Alert setAlert={setAlert} />)

    fireEvent.click(screen.getByRole('button'))
    expect(setAlert).toBeCalled()
  })
})
