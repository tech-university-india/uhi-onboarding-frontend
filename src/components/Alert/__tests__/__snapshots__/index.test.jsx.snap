// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Alert Dialog should render the component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-red-100 border border-red-400 text-red-700 px-4 py-3 rounded relative mt-2"
        role="alert"
      >
        <strong
          class="font-bold"
        >
          Alert
        </strong>
        <span
          class="block sm:inline ml-4"
        />
        <span
          class="absolute top-0 bottom-0 right-0 px-4 py-3 "
        >
          <svg
            class="fill-current h-6 w-6 text-red-500 "
            role="button"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Close
            </title>
            <path
              d="M14.348 14.849a1.2 1.2 0 0 1-1.697 0L10 11.819l-2.651 3.029a1.2 1.2 0 1 1-1.697-1.697l2.758-3.15-2.759-3.152a1.2 1.2 0 1 1 1.697-1.697L10 8.183l2.651-3.031a1.2 1.2 0 1 1 1.697 1.697l-2.758 3.152 2.758 3.15a1.2 1.2 0 0 1 0 1.698z"
            />
          </svg>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-red-100 border border-red-400 text-red-700 px-4 py-3 rounded relative mt-2"
      role="alert"
    >
      <strong
        class="font-bold"
      >
        Alert
      </strong>
      <span
        class="block sm:inline ml-4"
      />
      <span
        class="absolute top-0 bottom-0 right-0 px-4 py-3 "
      >
        <svg
          class="fill-current h-6 w-6 text-red-500 "
          role="button"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Close
          </title>
          <path
            d="M14.348 14.849a1.2 1.2 0 0 1-1.697 0L10 11.819l-2.651 3.029a1.2 1.2 0 1 1-1.697-1.697l2.758-3.15-2.759-3.152a1.2 1.2 0 1 1 1.697-1.697L10 8.183l2.651-3.031a1.2 1.2 0 1 1 1.697 1.697l-2.758 3.152 2.758 3.15a1.2 1.2 0 0 1 0 1.698z"
          />
        </svg>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
